modules/siddhi-core/src/main/java/io/siddhi/core/table/CacheTableLRU.java

```diff
@@ -77,15 +78,15 @@ public void deleteEntriesUsingCachePolicy(int numRowsToDelete) {
             indexEventHolder.deleteAll();
         } else {
             Set<Object> keys = indexEventHolder.getAllPrimaryKeyValues();
-            TreeMap<Long, Object> toDelete = new TreeMap<>();
+            Map<Long, Object> toDelete = new TreeMap<>();
```

Comment:
I don't this in this case (within a function we need to use Map) because this use case  specifically needs TreeMap

Reply:
Reverted back in https://github.com/siddhi-io/siddhi/pull/1579/commits/0c7f80dd9a9417a5d11619593de326278718e403